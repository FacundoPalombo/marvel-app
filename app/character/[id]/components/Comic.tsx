import styles from "./Comic.module.css";

type ComicProps = {
  img: string;
  title: string;
  issueDate: string;
};

export default function Comic({ img, title, issueDate }: ComicProps) {
  return (
    <div className={styles.comic}>
      <img className={styles.comic__image} src={img} alt={title} />
      <h3 className={styles.comic__title}>{title}</h3>
      <span className={styles.comic__issueDate}>{issueDate}</span>
    </div>
  );
}
