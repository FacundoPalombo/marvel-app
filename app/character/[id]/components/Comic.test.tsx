import { render } from "@testing-library/react";
import Comic from "./Comic";

describe("Comic unit tests", () => {
  it("should match to snapshot", () => {
    const props = {
      img: "image",
      title: "Tremendo comic",
      issueDate: "2024",
    };

    const { asFragment } = render(<Comic {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render ok", () => {
    const props = {
      img: "image",
      title: "Tremendo comic",
      issueDate: "2024",
    };

    const { getByRole } = render(<Comic {...props} />);

    expect(
      getByRole("heading", { name: "Tremendo comic" })
    ).toBeInTheDocument();
  });
});
