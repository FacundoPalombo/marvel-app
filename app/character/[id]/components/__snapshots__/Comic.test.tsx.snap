// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comic unit tests should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="comic"
  >
    <img
      alt="Tremendo comic"
      class="comic__image"
      src="image"
    />
    <h3
      class="comic__title"
    >
      Tremendo comic
    </h3>
    <span
      class="comic__issueDate"
    >
      2024
    </span>
  </div>
</DocumentFragment>
`;
